### DISCUSSION TITLE
Extending the data model

### DISCUSSION THREAD

**Discussion by alahiff** [SciCatProject/scicat-backend-next#1391] [General]
Are there any recommendations or guidelines for people wanting to use SciCat but who need to extend the data model? I've looked around but haven't found this mentioned anywhere. For example, we would like to have a **campaign**, which is similar to a proposal in that it has its own metadata and is associated with a group of datasets. As a proof-of-concept I tried adding campaigns to the backend by copying `src/proposals`, modifying everything as needed, as well as modifying some files in `src/casl`. This seems to have worked and wasn't too difficult, but before modying the frontend to also support campaigns we thought it would be good to check whether going down this road is the right thing to do or not.

**Comment by sbliven** [comment]
I think we do try to make sure that scicat reflects widely-used features and doesn't include too much that is specific to individual facilities. Can you describe your campaign use case in more detail? Maybe it can be implemented as a collection of proposals or a collection of datasets (eg as proposed in #805).

Once we have a clearer idea what a campaign involves I think we would either encourage you to open a detailed issue with the proposed change, or else if it's very facility specific we would suggest implementing it as a microservice run alongside scicat.
